File: vs2003_cd02/Program Files/Microsoft Visual Studio .NET 2003/SDK/v1.1/QuickStart/winforms/samples/writingcontrols/usercontrol/cs/control/customercontrol.cs
//-----------------------------------------------------------------------
//  This file is part of the Microsoft .NET SDK Code Samples.
// 
//  Copyright (C) Microsoft Corporation.  All rights reserved.
// 
//This source code is intended only as a supplement to Microsoft
//Development Tools and/or on-line documentation.  See these other
//materials for detailed information regarding Microsoft code samples.
// 
//THIS CODE AND INFORMATION ARE PROVIDED AS IS WITHOUT WARRANTY OF ANY
//KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
//IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
//PARTICULAR PURPOSE.
//-----------------------------------------------------------------------


namespace Microsoft.Samples.WinForms.Cs.CustomerControl {
	using System;
	using System.Windows.Forms;
	using System.Drawing;

	public class CustomerControl : System.Windows.Forms.UserControl {
		/// <summary>
		///    Required designer variable.
		/// </summary>
		private System.ComponentModel.Container components;
		protected internal System.Windows.Forms.TextBox textBoxTitle;
		protected internal System.Windows.Forms.Label labelTitle;
		protected internal System.Windows.Forms.TextBox textBoxAddress;
		protected internal System.Windows.Forms.TextBox textBoxLastName;
		protected internal System.Windows.Forms.TextBox textBoxFirstName;
		protected internal System.Windows.Forms.TextBox textBoxID;
		protected internal System.Windows.Forms.Label labelAddress;
		protected internal System.Windows.Forms.Label labelLastName;
		protected internal System.Windows.Forms.Label labelFirstName;
		protected internal System.Windows.Forms.Label labelID;

		private Microsoft.Samples.WinForms.Cs.CustomerControl.Customer customer1;

        public CustomerControl() {
			//
			// Required for Windows Form Designer support
			//
			InitializeComponent();

        }

        public Customer Customer {
            get {
                return customer1;
            }
            set {
				if (value == null)
				{
					customer1 = Microsoft.Samples.WinForms.Cs.CustomerControl.Customer.ReadCustomer();
				}
				else
				{
					customer1 = value;
				}
				LoadCustomer();
            }
        }

        public void AcceptChanges() {
            customer1.Title = textBoxTitle.Text;
            customer1.FirstName = textBoxFirstName.Text;
            customer1.LastName = textBoxLastName.Text;
            customer1.Address = textBoxAddress.Text;
        }

        public void RejectChanges() {
            LoadCustomer();
        }

        private void LoadCustomer() {
            textBoxID.Text = customer1.ID;
            textBoxTitle.Text = customer1.Title;
            textBoxFirstName.Text = customer1.FirstName;
            textBoxLastName.Text = customer1.LastName;
            textBoxAddress.Text =  customer1.Address;
        }

		/// <summary>
		///    Clean up any resources being used.
		/// </summary>
        protected override void Dispose(bool disposing)
        {
           if (disposing) {
                if (components != null) {
                    components.Dispose();
                }
           }
           base.Dispose(disposing);
        }

		/// <summary>
		///    Required method for Designer support - do not modify
		///    the contents of this method with the code editor.
		/// </summary>
		private void InitializeComponent() {
			this.components = new System.ComponentModel.Container ();
			this.labelAddress = new System.Windows.Forms.Label ();
			this.labelLastName = new System.Windows.Forms.Label ();
			this.textBoxFirstName = new System.Windows.Forms.TextBox ();
			this.textBoxLastName = new System.Windows.Forms.TextBox ();
			this.labelFirstName = new System.Windows.Forms.Label ();
			this.labelTitle = new System.Windows.Forms.Label ();
			this.textBoxTitle = new System.Windows.Forms.TextBox ();
			this.textBoxAddress = new System.Windows.Forms.TextBox ();
			this.labelID = new System.Windows.Forms.Label ();
			this.textBoxID = new System.Windows.Forms.TextBox ();
			this.Text = "CustomerControl";
			this.Size = new System.Drawing.Size (384, 304);
			labelAddress.Location = new System.Drawing.Point (8, 194);
			labelAddress.TabIndex = 9;
			labelAddress.Size = new System.Drawing.Size (64, 16);
			labelAddress.Text = "Address:";
			labelLastName.Location = new System.Drawing.Point (8, 154);
			labelLastName.TabIndex = 8;
			labelLastName.Size = new System.Drawing.Size (64, 16);
			labelLastName.Text = "LastName:";
			textBoxFirstName.Location = new System.Drawing.Point (88, 112);
			textBoxFirstName.TabIndex = 2;
			textBoxFirstName.Anchor = (System.Windows.Forms.AnchorStyles.Top | System.Windows.Forms.AnchorStyles.Left | System.Windows.Forms.AnchorStyles.Right);
			textBoxFirstName.Size = new System.Drawing.Size (240, 20);
			textBoxLastName.Location = new System.Drawing.Point (88, 152);
			textBoxLastName.TabIndex = 3;
			textBoxLastName.Anchor = (System.Windows.Forms.AnchorStyles.Top | System.Windows.Forms.AnchorStyles.Left | System.Windows.Forms.AnchorStyles.Right);
			textBoxLastName.Size = new System.Drawing.Size (240, 20);
			labelFirstName.Location = new System.Drawing.Point (8, 112);
			labelFirstName.TabIndex = 7;
			labelFirstName.Size = new System.Drawing.Size (64, 16);
			labelFirstName.Text = "First Name:";
			labelTitle.Location = new System.Drawing.Point (8, 72);
			labelTitle.TabIndex = 6;
			labelTitle.Size = new System.Drawing.Size (64, 16);
			labelTitle.Text = "Title:";
			textBoxTitle.Location = new System.Drawing.Point (88, 70);
			textBoxTitle.TabIndex = 1;
			textBoxTitle.Anchor = (System.Windows.Forms.AnchorStyles.Top | System.Windows.Forms.AnchorStyles.Left | System.Windows.Forms.AnchorStyles.Right);
			textBoxTitle.Size = new System.Drawing.Size (88, 20);
			textBoxAddress.Location = new System.Drawing.Point (88, 192);
			textBoxAddress.Multiline = true;
			textBoxAddress.AcceptsReturn = true;
			textBoxAddress.TabIndex = 4;
			textBoxAddress.Anchor = (System.Windows.Forms.AnchorStyles.Top | System.Windows.Forms.AnchorStyles.Bottom | System.Windows.Forms.AnchorStyles.Left | System.Windows.Forms.AnchorStyles.Right);
			textBoxAddress.Size = new System.Drawing.Size (264, 96);
			labelID.Location = new System.Drawing.Point (8, 32);
			labelID.TabIndex = 5;
			labelID.Size = new System.Drawing.Size (64, 16);
			labelID.Text = "ID:";
			textBoxID.Location = new System.Drawing.Point (88, 32);
			textBoxID.Enabled = false;
			textBoxID.ReadOnly = true;
			textBoxID.TabIndex = 0;
			textBoxID.Anchor = (System.Windows.Forms.AnchorStyles.Top | System.Windows.Forms.AnchorStyles.Left | System.Windows.Forms.AnchorStyles.Right);
			textBoxID.Size = new System.Drawing.Size (200, 20);
			this.Controls.Add (this.textBoxTitle);
			this.Controls.Add (this.labelTitle);
			this.Controls.Add (this.textBoxAddress);
			this.Controls.Add (this.textBoxLastName);
			this.Controls.Add (this.textBoxFirstName);
			this.Controls.Add (this.textBoxID);
			this.Controls.Add (this.labelAddress);
			this.Controls.Add (this.labelLastName);
			this.Controls.Add (this.labelFirstName);
			this.Controls.Add (this.labelID);
		}
	}
}

File: vs2003_cd02/Program Files/Microsoft Visual Studio .NET 2003/SDK/v1.1/QuickStart/winforms/samples/writingcontrols/usercontrol/vb/control/CustomerControl.vb
'-----------------------------------------------------------------------
'  This file is part of the Microsoft .NET SDK Code Samples.
' 
'  Copyright (C) Microsoft Corporation.  All rights reserved.
' 
'This source code is intended only as a supplement to Microsoft
'Development Tools and/or on-line documentation.  See these other
'materials for detailed information regarding Microsoft code samples.
' 
'THIS CODE AND INFORMATION ARE PROVIDED AS IS WITHOUT WARRANTY OF ANY
'KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
'IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
'PARTICULAR PURPOSE.
'-----------------------------------------------------------------------


Imports System
Imports System.Windows.Forms
Imports System.Drawing


Namespace Microsoft.Samples.WinForms.VB.CustomerControl

    Public Class CustomerControl
        Inherits System.Windows.Forms.UserControl

        Private customer1 As Customer

        Public Sub New()

            MyBase.New()

            '  Required by the Windows Forms Designer
            InitializeComponent()

        End Sub

        Public Property Customer() As Customer
            Get
                Return customer1
            End Get

            Set(ByVal Value As Customer)
                customer1 = Value
                LoadCustomer()
            End Set
        End Property

        Public Sub AcceptChanges()
            customer1.Title = textBoxTitle.Text
            customer1.FirstName = textBoxFirstName.Text
            customer1.LastName = textBoxLastName.Text
            customer1.Address = textBoxAddress.Text
        End Sub

        Public Sub RejectChanges()
            LoadCustomer()
        End Sub

        Private Sub LoadCustomer()
            textBoxID.Text = customer1.ID
            textBoxTitle.Text = customer1.Title
            textBoxFirstName.Text = customer1.FirstName
            textBoxLastName.Text = customer1.LastName
            textBoxAddress.Text = customer1.Address
        End Sub

        Protected Overloads Overrides Sub Dispose(ByVal disposing As Boolean)
            If disposing Then
                If Not (components Is Nothing) Then
                    components.Dispose()
                End If
            End If
            MyBase.Dispose(disposing)
        End Sub

#Region " Windows Form Designer generated code "

        'Required by the Windows Form Designer
        Private components As System.ComponentModel.Container
        Private WithEvents textBoxTitle As System.Windows.Forms.TextBox
        Private WithEvents labelTitle As System.Windows.Forms.Label
        Private WithEvents textBoxAddress As System.Windows.Forms.TextBox
        Private WithEvents textBoxLastName As System.Windows.Forms.TextBox
        Private WithEvents textBoxFirstName As System.Windows.Forms.TextBox
        Private WithEvents textBoxID As System.Windows.Forms.TextBox
        Private WithEvents labelAddress As System.Windows.Forms.Label
        Private WithEvents labelLastName As System.Windows.Forms.Label
        Private WithEvents labelFirstName As System.Windows.Forms.Label
        Private WithEvents labelID As System.Windows.Forms.Label

        'NOTE: The following procedure is required by the Windows Form Designer
        'It can be modified using the Windows Form Designer.
        'Do not modify it using the code editor.
        Private Sub InitializeComponent()
            Me.components = New System.ComponentModel.Container()
            Me.labelAddress = New System.Windows.Forms.Label()
            Me.labelLastName = New System.Windows.Forms.Label()
            Me.textBoxFirstName = New System.Windows.Forms.TextBox()
            Me.textBoxLastName = New System.Windows.Forms.TextBox()
            Me.labelFirstName = New System.Windows.Forms.Label()
            Me.labelTitle = New System.Windows.Forms.Label()
            Me.textBoxTitle = New System.Windows.Forms.TextBox()
            Me.textBoxAddress = New System.Windows.Forms.TextBox()
            Me.labelID = New System.Windows.Forms.Label()
            Me.textBoxID = New System.Windows.Forms.TextBox()

            labelAddress.Size = New System.Drawing.Size(64, 16)
            labelAddress.Location = New System.Drawing.Point(8, 194)
            labelAddress.TabIndex = 9
            labelAddress.Text = "Address:"

            labelLastName.Size = New System.Drawing.Size(64, 16)
            labelLastName.Location = New System.Drawing.Point(8, 154)
            labelLastName.TabIndex = 8
            labelLastName.Text = "LastName:"

            textBoxFirstName.Anchor = (System.Windows.Forms.AnchorStyles.Top Or System.Windows.Forms.AnchorStyles.Left Or System.Windows.Forms.AnchorStyles.Right)
            textBoxFirstName.Text = ""
            textBoxFirstName.TabIndex = 2
            textBoxFirstName.Size = New System.Drawing.Size(240, 20)
            textBoxFirstName.Location = New System.Drawing.Point(88, 112)

            Me.Text = "CustomerControl"
            Me.Size = New System.Drawing.Size(384, 304)

            textBoxLastName.Anchor = (System.Windows.Forms.AnchorStyles.Top Or System.Windows.Forms.AnchorStyles.Left Or System.Windows.Forms.AnchorStyles.Right)
            textBoxLastName.Location = New System.Drawing.Point(88, 152)
            textBoxLastName.Text = ""
            textBoxLastName.TabIndex = 3
            textBoxLastName.Size = New System.Drawing.Size(240, 20)

            labelFirstName.Location = New System.Drawing.Point(8, 112)
            labelFirstName.Size = New System.Drawing.Size(64, 16)
            labelFirstName.TabIndex = 7
            labelFirstName.Text = "First Name:"

            labelTitle.Size = New System.Drawing.Size(64, 16)
            labelTitle.Location = New System.Drawing.Point(8, 72)
            labelTitle.TabIndex = 6
            labelTitle.Text = "Title:"

            textBoxTitle.Anchor = (System.Windows.Forms.AnchorStyles.Top Or System.Windows.Forms.AnchorStyles.Left Or System.Windows.Forms.AnchorStyles.Right)
            textBoxTitle.Text = ""
            textBoxTitle.TabIndex = 1
            textBoxTitle.Size = New System.Drawing.Size(88, 20)
            textBoxTitle.Location = New System.Drawing.Point(88, 70)

            textBoxAddress.Anchor = (System.Windows.Forms.AnchorStyles.Top Or System.Windows.Forms.AnchorStyles.Bottom Or System.Windows.Forms.AnchorStyles.Left Or System.Windows.Forms.AnchorStyles.Right)
            textBoxAddress.Text = ""
            textBoxAddress.Multiline = True
            textBoxAddress.TabIndex = 4
            textBoxAddress.Size = New System.Drawing.Size(264, 96)
            textBoxAddress.Location = New System.Drawing.Point(88, 192)
            textBoxAddress.AcceptsReturn = True

            labelID.Size = New System.Drawing.Size(64, 16)
            labelID.Location = New System.Drawing.Point(8, 32)
            labelID.TabIndex = 5
            labelID.Text = "ID:"

            textBoxID.Anchor = (System.Windows.Forms.AnchorStyles.Top Or System.Windows.Forms.AnchorStyles.Left Or System.Windows.Forms.AnchorStyles.Right)
            textBoxID.Text = ""
            textBoxID.ReadOnly = True
            textBoxID.TabIndex = 0
            textBoxID.Size = New System.Drawing.Size(200, 20)
            textBoxID.Location = New System.Drawing.Point(88, 30)
            textBoxID.Enabled = False

            Me.Controls.Add(textBoxTitle)
            Me.Controls.Add(labelTitle)
            Me.Controls.Add(textBoxAddress)
            Me.Controls.Add(textBoxLastName)
            Me.Controls.Add(textBoxFirstName)
            Me.Controls.Add(textBoxID)
            Me.Controls.Add(labelAddress)
            Me.Controls.Add(labelLastName)
            Me.Controls.Add(labelFirstName)
            Me.Controls.Add(labelID)

        End Sub

#End Region

    End Class ' CustomerControl

End Namespace ' Microsoft.Samples.WinForms.VB.CustomerControl
